### PR TITLE
Mkdocs and documentation deployment via GitHub action

### LINKED ISSUE #43: Created docs via mkdocs or some other package
Would be nice to have the basics on README file and a more detailed documentation somewhere else

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds
- Provides basic structure for the documentation files using mkdocs (I've just written the index, other docs to come in other PRs) - Fix #43
- Adds a GitHub action that will deploy docs on GitHub pages

### How to test:
- Run mkdocs serve

### Expected outcome:
- [ ] Docs pages should show up without errors

### Review:
- [x] Code approved by HS
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner

### CODE REVIEW

**Comment by northwestwitch** [comment]
Tested locally: 🆗 

image

File: .github/workflows/mkdocs_publishing.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: ci
```
Comment (henrikstranneheim) [inline .github/workflows/mkdocs_publishing.yml:1]:
```suggestion
name: Publish Mkdocs
```

File: .github/workflows/mkdocs_publishing.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          python-version: 3.x
+      - run: echo "cache_id=$(date --utc '+%V')" >> $GITHUB_ENV
+      - uses: actions/cache@v3
+        with:
+          key: mkdocs-material-${{ env.cache_id }}
+          path: .cache
+          restore-keys: |
+            mkdocs-material-
+      - run: pip install mkdocs-material
+      - run: mkdocs gh-deploy --force
```
Comment (henrikstranneheim) [inline .github/workflows/mkdocs_publishing.yml:24]:
```suggestion
      - run: mkdocs gh-deploy --force

```

File: docs/deployment/installation.md [hunk 3]
```diff
@@ -0,0 +1 @@
+## Installing Chanjo2 on a virtual environment
```
Comment (henrikstranneheim) [inline docs/deployment/installation.md:1]:
```suggestion
## Installing Chanjo2 in a virtual environment
```

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+The tool is flexible and can be used in different ways. The simplest use case would be calculating sequencing coverage over one or more intervals for a d4 file stored locally or remotely on the internet.
+
+## Chanjo2 image as a proxy to d4tools to compute coverage stats over genomic intervals of a d4 file
+
+Chanjo2 image contains [d4tools][d4tools-tool] and can be used to directly retrieve statistics over d4 files.
+
+* Executing d4tools:
+``` bash
```
Comment (henrikstranneheim) [inline docs/index.md:16]:
```suggestion

``` shell
```

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+Chanjo2 image contains [d4tools][d4tools-tool] and can be used to directly retrieve statistics over d4 files.
+
+* Executing d4tools:
+``` bash 
+docker run --entrypoint d4tools --rm  clinicalgenomics/chanjo2-stage:latest
+```
+
+* Coalculating coverage on specific genomic intervals of a d4 file using d4tools
+
+``` bash
```
Comment (henrikstranneheim) [inline docs/index.md:22]:
```suggestion
``` shell
```

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+single d4 files or samples stored in the database with associated d4 files.
+
+
+The tool is flexible and can be used in different ways. The simplest use case would be calculating sequencing coverage over one or more intervals for a d4 file stored locally or remotely on the internet.
+
+## Chanjo2 image as a proxy to d4tools to compute coverage stats over genomic intervals of a d4 file
+
+Chanjo2 image contains [d4tools][d4tools-tool] and can be used to directly retrieve statistics over d4 files.
+
+* Executing d4tools:
```
Comment (henrikstranneheim) [inline docs/index.md:15]:
```suggestion
### Executing d4tools
```

File: docs/index.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+``` bash 
+docker run --entrypoint d4tools --rm  clinicalgenomics/chanjo2-stage:latest
+```
+
+* Coalculating coverage on specific genomic intervals of a d4 file using d4tools
```
Comment (henrikstranneheim) [inline docs/index.md:20]:
```suggestion
### Calculating coverage on specific genomic intervals of a d4 file using d4tools
```

File: docs/usage/interval-queries.md [hunk 8]
```diff
@@ -0,0 +1 @@
+## retrieving available genetic intervals (genes, transcripts, exons) from database
```
Comment (henrikstranneheim) [inline docs/usage/interval-queries.md:1]:
```suggestion
## Retrieving available genetic intervals (genes, transcripts, exons) from database
```

File: mkdocs.yml [hunk 9]
```diff
@@ -0,0 +1,19 @@
+site_name: Chanjo2
+site_url: http://chanjo2.scilifelab.se
```
Comment (henrikstranneheim) [inline mkdocs.yml:2]: Should we not use https here?

**Review by henrikstranneheim — APPROVED** [review]
Great!

File: mkdocs.yml (hunk 9 above)
Comment (northwestwitch) [reply mkdocs.yml:2]: Yes, better with httpS

**Pull request merged**